server/src/test/java/io/druid/client/cache/CacheDistributionTest.java

```diff
@@ -0,0 +1,149 @@
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+
+@RunWith(Parameterized.class)
+public class CacheDistributionTest
+{
+  public static final int KEY_COUNT = 1_000_000;
+
+  @Parameterized.Parameters(name = "repetitions={0}, hash={1}")
+  public static Iterable<Object[]> data() {
+    List<HashAlgorithm> hash = ImmutableList.of(
+        DefaultHashAlgorithm.FNV1A_64_HASH, DefaultHashAlgorithm.KETAMA_HASH, MemcachedCache.MURMUR3_128
+    );
+    List<Integer> repetitions = Arrays.asList(160, 500, 1000, 2500, 5000);
+
+    Set<List<Object>> values = Sets.cartesianProduct(
+        Sets.newLinkedHashSet(hash),
+        Sets.newLinkedHashSet(repetitions)
+    );
+    return Iterables.transform(
+        values, new Function<List<Object>, Object[]>()
+        {
+          @Nullable
+          @Override
+          public Object[] apply(List<Object> input)
+          {
+            return input.toArray();
+          }
+        }
+    );
+  }
+
+  final HashAlgorithm hash;
+  final int reps;
+
+  @BeforeClass
+  public static void header() {
+    System.out.printf(
+        "%25s\t%5s\t%10s\t%10s\t%10s\t%10s\t%10s\t%7s\t%5s\n",
+        "hash", "reps", "node 1", "node 2", "node 3", "node 4", "node 5", "min/max", "ns"
+    );
+  }
+
+  public CacheDistributionTest(final HashAlgorithm hash, final int reps)
+  {
+    this.hash = hash;
+    this.reps = reps;
+  }
+
+  // run to get a sense of cache key distribution for different ketama reps / hash functions
+  @Test
+  public void testDistribution() throws Exception
+  {
+    KetamaNodeLocator locator = new KetamaNodeLocator(
+        ImmutableList.of(
+            dummyNode("druid-cache.0001", 11211),
+            dummyNode("druid-cache.0002", 11211),
+            dummyNode("druid-cache.0003", 11211),
+            dummyNode("druid-cache.0004", 11211),
+            dummyNode("druid-cache.0005", 11211)
+        ),
+        hash,
+        new DefaultKetamaNodeLocatorConfiguration()
+        {
+          @Override
+          public int getNodeRepetitions()
+          {
+            return reps;
+          }
+        }
+    );
+
+    Map<MemcachedNode, AtomicLong> counter = Maps.newHashMap();
+    long t = 0;
+    for(int i = 0; i < KEY_COUNT; ++i) {
+      final String k = DigestUtils.sha1Hex("abc" + i) + ":" + DigestUtils.sha1Hex("xyz" + i);
+      long t0 = System.nanoTime();
+      MemcachedNode node = locator.getPrimary(k);
+      t += System.nanoTime() - t0;
+      if(counter.containsKey(node)) {
+        counter.get(node).incrementAndGet();
+      } else {
+        counter.put(node, new AtomicLong(1));
+      }
+    }
+
+    long min = Long.MAX_VALUE;
+    long max = 0;
+    System.out.printf("%25s\t%5d\t", hash, reps);
+    for(AtomicLong count : counter.values()) {
+      System.out.printf("%10d\t", count.get());
```

Comment:
since we are only interested in distribution of keys, so it doesn't matter.
But, counter values ordering would not necessarily be node1..node5 each time, printed table makes it look like number of keys reported in a column is for that "particular" node. It can be corrected by using a sorted map at line 117.